src/modules/rover_mecanum/RoverMecanumControl/RoverMecanumControl.hpp

```diff
@@ -0,0 +1,137 @@
+ *
+ ****************************************************************************/
+
+#pragma once
+
+// PX4 includes
+#include <px4_platform_common/module_params.h>
+
+// uORB includes
+#include <uORB/Publication.hpp>
+#include <uORB/PublicationMulti.hpp>
+#include <uORB/Subscription.hpp>
+#include <uORB/topics/rover_mecanum_setpoint.h>
+#include <uORB/topics/rover_mecanum_status.h>
+#include <uORB/topics/actuator_motors.h>
+
+
+// Standard libraries
+#include <lib/pid/pid.h>
+#include <matrix/matrix/math.hpp>
+#include <matrix/math.hpp>
+#include <mathlib/mathlib.h>
+#include <math.h>
+
+using namespace matrix;
+
+/**
+ * @brief Class for mecanum rover guidance.
+ */
+class RoverMecanumControl : public ModuleParams
+{
+public:
+	/**
+	 * @brief Constructor for RoverMecanumControl.
+	 * @param parent The parent ModuleParams object.
+	 */
+	RoverMecanumControl(ModuleParams *parent);
+	~RoverMecanumControl() = default;
+
+	/**
+	 * @brief Compute motor commands based on setpoints
+	 * @param vehicle_yaw Yaw of the vehicle [rad]
+	 * @param vehicle_yaw_rate Yaw rate of the vehicle [rad/s]
+	 * @param vehicle_forward_speed Forward speed of the vehicle [m/s]
+	 * @param vehicle_lateral_speed Lateral speed of the vehicle [m/s]
+	 */
+	void computeMotorCommands(float vehicle_yaw, float vehicle_yaw_rate, float vehicle_forward_speed,
+				  float vehicle_lateral_speed);
+
+	/**
+	 * @brief Reset PID controllers
+	 */
+	void resetControllers();
+
+protected:
+	/**
+	 * @brief Update the parameters of the module.
+	 */
+	void updateParams() override;
+
+private:
+	/**
+	 * @brief Turn normalized speed setpoints into normalized motor commands.
+	 *
+	 * @param forward_speed Normalized linear speed in body forward direction [-1, 1].
+	 * @param lateral_speed Normalized linear speed in body lateral direction [-1, 1].
+	 * @param speed_diff Normalized speed difference between left and right wheels [-1, 1].
+	 * @return matrix::Vector4f: Normalized motor inputs [-1, 1]
+	 */
+	matrix::Vector4f computeInverseKinematics(float forward_speed, float lateral_speed, float speed_diff);
+
+	// uORB subscriptions
+	uORB::Subscription _rover_mecanum_setpoint_sub{ORB_ID(rover_mecanum_setpoint)};
+
+	// uORB publications
+	uORB::PublicationMulti<actuator_motors_s> _actuator_motors_pub{ORB_ID(actuator_motors)};
+	uORB::Publication<rover_mecanum_status_s> _rover_mecanum_status_pub{ORB_ID(rover_mecanum_status)};
+
+	// Variables
+	rover_mecanum_setpoint_s _rover_mecanum_setpoint{};
+	hrt_abstime _timestamp{0};
+	float _max_yaw_rate{0.f};
+
+	// Controllers
+	PID_t _pid_forward_throttle; // PID for the closed loop forward speed control
+	PID_t _pid_lateral_throttle; // PID for the closed loop lateral speed control
+	PID_t _pid_yaw; // PID for the closed loop yaw control
+	PID_t _pid_yaw_rate; // PID for the closed loop yaw rate control
+
+	// Constants
+	static constexpr float YAW_RATE_ERROR_THRESHOLD = 1.f; // [rad/s] Error threshold for the closed loop yaw rate control
```

Comment:
improve comment - what is happening at this threshold?

Reply:
This used to be a cut off for the closed loop yaw rate controller, but this has been replaced with the thresholds on the measurements.
Removed the parameter in [7ac0ef0](https://github.com/PX4/PX4-Autopilot/pull/23708/commits/7ac0ef094be37f398df2e07c2c465ac3e5e3f56f)